### ISSUE TITLE
com.google.cloud.spark.bigquery.repackaged.com.google.common.io.BaseEncoding$DecodingException: Invalid input length 25

### ISSUE THREAD

**Issue by kumgaurav** [GoogleCloudPlatform/spark-bigquery-connector#90]
When I am running locally I can connect to bigquery but running on dataproc is throwing below error
Exception in thread "main" java.lang.IllegalArgumentException: com.google.cloud.spark.bigquery.repackaged.com.google.common.io.BaseEncoding$DecodingException: Invalid input length 25
	at com.google.cloud.spark.bigquery.repackaged.com.google.common.io.BaseEncoding.decode(BaseEncoding.java:219)
	at com.google.cloud.spark.bigquery.repackaged.com.google.api.client.util.Base64.decodeBase64(Base64.java:104)
	at com.google.cloud.spark.bigquery.SparkBigQueryOptions.createCredentials(SparkBigQueryOptions.scala:47)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider$.createBigQuery(BigQueryRelationProvider.scala:125)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider$$anonfun$getOrCreateBigQuery$1.apply(BigQueryRelationProvider.scala:107)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider$$anonfun$getOrCreateBigQuery$1.apply(BigQueryRelationProvider.scala:107)
	at scala.Option.getOrElse(Option.scala:121)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.getOrCreateBigQuery(BigQueryRelationProvider.scala:107)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelationInternal(BigQueryRelationProvider.scala:53)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelation(BigQueryRelationProvider.scala:37)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:341)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:239)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:227)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:164)
	at com.paloaltonetworks.GenarateCorruptIdForCleaning$.main(GenarateCorruptIdForCleaning.scala:41)

**Comment by kumgaurav** [comment]
Using the spark conf method fixed the issue and I am able to read it now.
val sparkConf = new SparkConf().setAppName("Test").set("credentialsFile", credentialsStreamFilePath)

setting credential during table read is not working.

**Comment by megancooper** [comment]
@kumgaurav Could you provide some more code detailing what `credentialsStreamFilePath ` is? I am trying to get this to work in an AWS EMR cluster, and am still not able to set credentials to the correct path.

**Comment by kumgaurav** [comment]
while you are submitting your spark job use this argument to submit the job "--files gs://cred/abc.json" , then your cred file will be get downloaded to executors and you just need to mention name of the cred file in your program.

let me know if it helps.

**Comment by megancooper** [comment]
@kumgaurav Ok thanks I will try this. I have attempted this before and it failed; I am running this in an AWS EMR cluster and my json file is located at: `/home/hadoop/credentials.json`. For some reason when I run `spark-submit --files /home/hadoop/credentials.json ...[other configs]...` I still end up with a `FileNotFound` error when trying to access it as `credentials.json` inside my script.

I am kind of lost and don't know how I can possibly authenticate from emr...Additionally, I have tried to use the `base64` option, but [I end up with an encoding error](https://stackoverflow.com/questions/59258549/spark-bigquery-connector-baseencodingdecodingexception-unrecognized-character). I know that my `base64` string is correct because I have decoded it, and checked it online in validators.
